### PR TITLE
Add CodeCov GitHub action and badge.

### PR DESCRIPTION
**Pull request by tcnghia**
# Description

* Add JaCoCo plugin
* Add CodeCov GitHub action and badge.

## Issue reference

Please reference the issue this PR will close: https://github.com/dapr/dapr/issues/2069 

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [ ] Created/updated tests
* [x] Extended the documentation

### CODE REVIEW

**Comment by mukundansundar** [comment]
@tcnghia ok to test is for dapr/dapr repo. Here the build already builds, test and runs IT for the code.

File: .codecov.yaml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+ignore:
+  - examples
```
Comment (mukundansundar) [inline .codecov.yaml:2]: ignore the sdk-tests module completely?

Comment (tcnghia) [reply .codecov.yaml:2]: Done

**Comment by tcnghia** [comment]
@mukundansundar @artursouza I've updated based on the review feedbacks. Can you please help take another look? Thanks

**Review by mukundansundar — APPROVED** [review]
lgtm. @artursouza ?

**Review by artursouza — APPROVED** [review]

**Comment by tcnghia** [comment]
Thanks @mukundansundar @artursouza !

**Pull request merged**